src/components/alerts/WalletAlert.tsx

```diff
@@ -60,28 +104,123 @@ const WalletAlert = ({ isOpen, onCancel, refetch }: Props) => {
           ) {
             toast({
               title: "Error",
-              description: response.response.data.error,
+              description:
+                response.response?.data?.message || "Something went wrong!",
               status: "error",
               duration: 9000,
               isClosable: true,
             });
             return;
           }
-
-          handleClose();
-          refetch();
-          toast({
-            title: "Paid",
-            description: "We've paid your invoice. Keep reviewing!",
-            status: "success",
-            duration: 9000,
-            isClosable: true,
-          });
+          if (response.metadata) {
+            const metadataArray = JSON.parse(response.metadata);
+            const addressPosition =
+              metadataArray
+                .flat()
+                .findIndex((elm: string) => elm.includes("text/identifier")) +
+              1;
+            if (metadataArray.flat()[addressPosition] === invoiceInput) {
+              setLightningData(response as LightningResponse);
+              setStep(2);
+            }
+            return;
+          }
         },
       }
     );
   };
 
+  const handleWithdrawAmount = (e: ChangeEvent<HTMLInputElement>) => {
+    let sendingAmount = e.target.value;
+
+    if (sendingAmount.match(/^[0-9]+$/)) {
+      // turn into a number type
+      setAmountToSend(Number(sendingAmount));
+    } else {
+      if (sendingAmount.length === 0) {
+        setAmountToSend(undefined);
+      }
+    }
+  };
+
+  const handleWithdrawalValidation = () => {
+    const amountToSendParsed = Number(amountToSend || 0);
+    const isWithinBalance = +amountToSendParsed <= +balance;
+    const minimumSendable = (lightningData?.minSendable || 1000) / 1000;
+    const maximumSendable = (lightningData?.maxSendable || 1000) / 1000;
+    const isWithinSendable =
+      Number(amountToSendParsed) >= minimumSendable &&
+      amountToSendParsed <= maximumSendable;
+    setError("");
+    if (!isWithinBalance) {
+      setError("Error: send sats lower than balance");
+      return;
+    }
+
+    if (!isWithinSendable) {
+      setError("Error: send sats within range");
```

Comment:
Within range of what? this error message is not descriptive enough
```suggestion
      setError("Error: send sats within range of <insert minSendable> and <insert maxSendable>");
```